torch_xla/distributed/xla_dist.py

```diff
@@ -86,32 +104,56 @@ def __init__(self,
                ' will interfere with the values set for distributed'
                ' training'.format(dist_var)))
 
+  def _check_client_mesh_health(
+      self, uneven_health_timeout=300, even_health_timeout=1800):
+    max_delay = 0.0
+    count = None
+    now = time.time()
+    for cw_hb in self._last_heartbeats.values():
+      max_delay = max(max_delay, now - cw_hb['last_time'])
+      if count is None:
+        count = cw_hb['count']
```

Comment:
I just realized this is per host, while mark_step prints are per core.
Should be fine anyway.